common/network_manager.go

```diff
@@ -179,9 +179,27 @@ func parseTTL(ttlS string) (time.Duration, error) {
 }
 
 func (m *NetworkManager) deleteRequestByID(reqID network.RequestID) {
+	var req *Request
 	m.reqsMu.Lock()
-	defer m.reqsMu.Unlock()
+	req = m.reqIDToRequest[reqID]
 	delete(m.reqIDToRequest, reqID)
+	m.reqsMu.Unlock()
+
+	// We need to nil the request in response otherwise they hold onto each
+	// others reference preventing the GC from cleaning the memory up.
+	req.responseMu.Lock()
+	if req.response != nil {
+		req.response.request = nil
```

Comment:
This was the main issue causing the memory leak.

Reply:
Also leaks in `browser.conn` which requires it to be `nil`ed when the iteration ends, and other such objects needs to be `nil`ed when the iteration ends.

Reply:
I have moved the commits to explore the other memory leaks into a new branch and [PR](https://github.com/grafana/xk6-browser/pull/1484).